Added C++ language

anjaliSilawat opened the pull request:
### Description 
I have expanded the language section by adding comprehensive content on C++. It covers various topics such as data types, control structures, functions, classes, object-oriented programming, and memory management. The content is suitable for both beginners and advanced learners.

issue #629 

![image](https://github.com/rupali-codes/LinksHub/assets/108583230/8c3421b0-481b-4d3d-9659-a410c8694498)


### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

rupali-codes reviewed (changes requested):
I have addressed a few things:
* please update your code with `linkshub:main`
* the description should be at least 15 words long
* your changes are not reflected on the site, please read & follow our [CONTRIBUTING guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md) carefully

anjaliSilawat commented:
> I have addressed a few things:
> 
> * please update your code with `linkshub:main`
> * the description should be at least 15 words long
> * your changes are not reflected on the site, please read & follow our [CONTRIBUTING guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md) carefully

@rupali-codes  I have updated the description also I update my code hope so changes are visible now on the site

rupali-codes commented:
@anjaliSilawat hey, please resolve the merge conflicts and your vercel deployment is getting bcz the of the following issues, kindly fix them
![image](https://github.com/rupali-codes/LinksHub/assets/78981177/86a90cff-bc76-489f-8de6-90df4c31633f)

CBID2 commented:
@anjaliSilawat, can you mention the issue number that your PR solves per the [Contributing files](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md#making-pull-requests) please?

anjaliSilawat commented:
> @anjaliSilawat, can you mention the issue number that your PR solves per the [Contributing files](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md#making-pull-requests) please?

done

rupali-codes commented:
@anjaliSilawat can you please resolve merge conflicts

anjaliSilawat commented:
> @anjaliSilawat can you please resolve merge conflicts


Yes, I'll give me a little bit of time. @rupali-codes

rupali-codes commented:
> > @anjaliSilawat can you please resolve merge conflicts
> 
> Yes, I'll give me a little bit of time. @rupali-codes

Sure, take your time

rupali-codes commented:
@anjaliSilawat are you ready to work on it now?

CBID2 commented on database/languages/c++.json:
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "name": "c++ Tutorial",
+        "description": " Learn the basics of C++ it is a popular programming language.C++ is used to create computer programs, and is one of the most used language in game development.",
```
[Recommendation] This description mostly describes what C++ is. While helpful, I highly suggest describing **why** people in tech should use this source. 
```suggestion
        "description": " Learn the basics of C++ it is a popular programming language.C++ is used to create computer programs, and is one of the most used languages in game development.",
```

CBID2 commented on database/languages/c++.json:
```diff
@@ -0,0 +1,25 @@
+        "subcategory": "C++"
+    },
+    {
+        "name": "C++ Recent Articles! ",
+        "description": "Recent Articles on C++",
+        "url": "https://www.geeksforgeeks.org/category/cpp/",
+        "category": "Languages",
+        "subcategory": "C++"
+    },
+    {
+        "name": "C++ Interview Questions",
+        "description": "Top 50 C++ Interview Questions and Answers (2023)",
+        "url": "https://www.geeksforgeeks.org/cpp-interview-questions/",
+        "category": "Languages",
+        "subcategory": "C++"
```
[Recommendation] Use my feedback in the [previous lines](https://github.com/rupali-codes/LinksHub/pull/957/files#r1234652002) to improve these descriptions. 
```suggestion
        "name": "C++ Recent Articles! ",
        "description": "Recent Articles on C++",
        "url": "https://www.geeksforgeeks.org/category/cpp/",
        "category": "Languages",
        "subcategory": "C++"
    },
    {
        "name": "C++ Interview Questions",
        "description": "Top 50 C++ Interview Questions and Answers (2023)",
        "url": "https://www.geeksforgeeks.org/cpp-interview-questions/",
        "category": "Languages",
        "subcategory": "C++"
```

CBID2 reviewed (changes requested):
Hi @anjaliSilawat! :) I resolved the merge conflicts from your branch. What's left to do now is revising the descriptions per my advice in my in-line comments(keep this advice in mind as you submit more pull requests to LinksHub).

rupali-codes commented:
@anjaliSilawat i'm closing this PR bcz you have resolved conflicts improperly, please make fresh PR and before that make sure to update your branch with `linkshub:main`.
